docs/docs/cmd/search/search.mdx

```diff
@@ -0,0 +1,248 @@
+﻿import Global from '/docs/cmd/_global.mdx';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# search
+
+Uses the Microsoft Search to query Microsoft 365 data
+
+## Usage
+
+```sh
+m365 search [options]
+```
+
+## Options
+
+```md definition-list
+`-q --queryText [queryText]`
+: Query to be executed in Keyword Query Language (KQL) format.
+
+`-s, --scopes <scopes>`
+: Comma-separated list of scopes for the search request. Allowed values are `chatMessage`, `message`, `event`, `drive`, `driveItem`, `list`, `listItem`, `site`, `bookmark`, `acronym`, or `person`.
+
+`--startIndex [startIndex]`
+: Set the number that indicates the 0-based starting point to list search results on the page. The default value is 0.
+
+`--pageSize [pageSize]`
+: Set the number that indicates the number of results to be returned for a page. The default is 25 results. The maximum is 500 results.
+
+`--allResults`
+: Set to get all search results from all pages.
+
+`--resultsOnly`
+: Set to get only the results form the query response.
+
+`--enableTopResults`
+: Use to get the first three messages in the response are sorted by relevance. Use this option only when scopes is set to message type.
+
+`--select [select]`
+: Comma-separated list of properties to return in the search results.
+
+`--sortBy [sortBy]`
+: Comma-separated list of properties to sort search results.
+
+`--enableSpellingSuggestion`
+: Set to get the suggested spelling correction information for typos in the user query.
+
+`--enableSpellingModification`
+: Set to get the search results for the corrected query when there are no results for the original query with typos, and get the corresponding correction information.
+```
+
+<Global />
+
+## Remarks
+
+The `queryText` option supports [Keyword Query Language](https://learn.microsoft.com/graph/api/resources/search-api-overview#keyword-query-language-kql-support) (KQL).
+
+The `sortBy` option allows to specify the order. By default, the order is ascending. If you want to apply descending order, add `:desc` to the property name, like `createdDateTime:desc`.
+
+## Examples
+
+Queries messages in the current user mailbox that contain the string "contoso" in any part of the message (the sender name, subject, message body, or any attachments). The query returns the first 100 results. The search results are ordered by createdDateTime descending and id descending and date ascending
+
+```sh
+m365 search --scopes 'message' --queryText 'contoso' --size 100 --sortBy 'createdDateTime:desc,id:desc,date'
```

Comment:
wrong option name

```suggestion
m365 search --scopes 'message' --queryText 'contoso' --pageSize 100 --sortBy 'createdDateTime:desc,id:desc,date'
```

Reply:
also I think sorting wont work with `message` type

Reply:
copy/paste from the spec